src/fits/fitsbody.c

```diff
@@ -344,7 +443,8 @@ void	read_body(tabstruct *tab, PIXTYPE *ptr, size_t size)
         if (spoonful>size)
           spoonful = size;
         bufdata = (char *)bufdata0;
-        QFREAD(bufdata, spoonful*tab->bytepix, cat->file, cat->filename);
+        readSpoonful(tab, spoonful, bufdata0, bufdata);
```

Comment:
Here `bufdata0` is of type `int *` but the function expects `'double *`. Not sure if that's a problem.